adapters/oidc/adapter-core/src/main/java/org/keycloak/adapters/HttpClientBuilder.java

```diff
@@ -287,16 +289,20 @@ public HttpClient build() {
                 cm = new SingleClientConnManager(registry);
             }
             BasicHttpParams params = new BasicHttpParams();
-            if (socketTimeout > -1)
-            {
+
+            if (proxyHost != null) {
+                params.setParameter(ConnRoutePNames.DEFAULT_PROXY, proxyHost);
```

Comment:
What about the following from section 2.8 in https://hc.apache.org/httpcomponents-client-ga/tutorial/html/connmgmt.html:
`
HttpHost proxy = new HttpHost("someproxy", 8080);
DefaultProxyRoutePlanner routePlanner = new DefaultProxyRoutePlanner(proxy);
CloseableHttpClient httpclient = HttpClients.custom()
        .setRoutePlanner(routePlanner)
        .build();
`